packages/opds-web-client/src/components/DownloadButton.tsx

```diff
@@ -12,14 +12,14 @@ export interface DownloadButtonProps extends React.HTMLProps<any> {
 }
 
 /** Shows a button to fulfill and download a book or download it directly. */
-export default class DownloadButton extends React.Component<DownloadButtonProps, void> {
+export default class DownloadButton extends React.Component<DownloadButtonProps, {}> {
   constructor(props) {
     super(props);
     this.fulfill = this.fulfill.bind(this);
   }
 
   render() {
-    let props = Object.assign({}, this.props);
+    let props = JSON.parse(JSON.stringify(this.props));
```

Comment:
This one doesn't seem to have props that aren't stringifiable but I'm not sure what all the HTML props are.

Reply:
If I change it to destructure and use `...rest` and `Object.assign`, I get errors that the props variable is not compatible with the anchor element's prop interface because it is missing properties. I don't know how to fix that just yet.